lib/srv/reexec.go

```diff
@@ -721,6 +745,63 @@ func checkHomeDir(localUser *user.User) (bool, error) {
 	return true, nil
 }
 
+type parker struct {
+	cmd  *exec.Cmd
+	pipe *os.File
+}
+
+func (p *parker) Close() error {
+	if p.cmd != nil {
+		p.cmd.Process.Kill()
+		p.cmd.Wait()
+		p.cmd = nil
+	}
+	if p.pipe != nil {
+		p.pipe.Close()
+		p.pipe = nil
+	}
+	return nil
+}
+
+// Spawns a process with the given credentials; uses `teleport forward`
+// internally, which will block reading from the pipe at FD 3. Returns a
+// io.Closer that should be closed when the parked process is no longer needed.
+func newParker(credential syscall.Credential) (io.Closer, error) {
+	executable, err := os.Executable()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	rp, wp, err := os.Pipe()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	defer rp.Close()
+
+	cmd := &exec.Cmd{
+		Path: executable,
+		Args: []string{executable, teleport.ForwardSubCommand},
```

Comment:
@espadolini @lxea I'm not fully sure why we're using "teleport forward" command here TBH which is used for port-forwarding. Would it not make more sense to implement another helper teleport subcommand that would block on reading from FD than trying to reuse an existing one that kinda does what we want but was really designed for another purpose?